### PR TITLE
infisical: init at 0.3.7

### PR DESCRIPTION
**Pull request by ivanmoreau**
###### Description of changes

Adds a new package, Infisical, an Open Source, End-to-End encrypted platform that enables secure syncing of env secrets and configs across teams, devices, and infrastructure.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by fabianhjr — APPROVED** [review]
Using nixpkgs-review:

1 package built:
infisical

```
[nix-shell:~/.cache/nixpkgs-review/pr-225258]$ infisical 
Infisical is a simple, end-to-end encrypted service that enables teams to sync and manage their environment variables across their development life cycle.

Usage:
  infisical [command]

Available Commands:
  export      Used to export environment variables to a file
  help        Help about any command
  init        Used to initialize your project with Infisical
  login       Login into your Infisical account
  reset       Used delete all Infisical related data on your machine
  run         Used to inject environments variables into your application process
  secrets     Used to create, read update and delete secrets
  vault       Used to manage where your Infisical login token is saved on your machine

Flags:
  -d, --debug           Enable verbose logging
      --domain string   Point the CLI to your own backend [can also set via environment variable name: INFISICAL_API_URL] (default "https://app.infisical.com/api")
  -h, --help            help for infisical
  -t, --toggle          Help message for toggle
  -v, --version         version for infisical

Use "infisical [command] --help" for more information about a command.
```

Builds and basic cli invocation works.

Thanks for the contribution! UwU

File: pkgs/development/tools/infisical/default.nix [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      longDescription = ''
+        Infisical is an Open Source, End-to-End encrypted platform that lets you
+        securely sync secrets and configs across your team, devices, and infrastructure
+      '';
+      mainProgram = "infisical";
+      homepage = "https://infisical.com/";
+      downloadPage = "https://github.com/Infisical/infisical/releases/";
+      license = licenses.mit;
+      maintainers = [ maintainers.ivanmoreau ];
+      platforms = [ "x86_64-linux" "x86_64-darwin" "aarch64-darwin" "aarch64-linux" ];
```
Comment (SuperSandro2000) [inline pkgs/development/tools/infisical/default.nix:59]:
```suggestion
      platforms = platforms.unix;
```
any reason why it shouldn't work for other linux arches?

File: pkgs/development/tools/infisical/default.nix [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      inherit sha256;
+    };
+
+    sourceRoot = ".";
+    installPhase = ''
+      mkdir -p $out/bin/
+      mkdir -p $out/share/completions/
+      mkdir -p $out/share/man/
+      cp completions/* $out/share/completions/
+      cp manpages/* $out/share/man/
```
Comment (SuperSandro2000) [inline pkgs/development/tools/infisical/default.nix:38]: Why isn't installManPage and installShellCompletion used here, too?

File: pkgs/development/tools/infisical/default.nix (hunk 1 above)
Comment (fabianhjr) [reply pkgs/development/tools/infisical/default.nix:59]: Per platform releases:

```
   x86_64-linux = "194akxb692xpqppakw49aywp5ma43yfcwv5imw4pm05cna0n06b1";
    x86_64-darwin = "0bgjx54c00v0nb88rzdv09g92yw9qsf2fxd8565g6fsw591va1pa";
    aarch64-linux = "0z07aikjhk9055apbvyaxdp8cgjl291fqgwgfbp9y3826q7s0riq";
    aarch64-darwin = "0garlx458jy6dpqbfd0y2p7xj9hagm815cflybbbxf5yz2v9da01";
```

Unsure if packaged upstream for other unixes but would probably need manual support.

File: pkgs/development/tools/infisical/default.nix (hunk 2 above)
Comment (fabianhjr) [reply pkgs/development/tools/infisical/default.nix:38]: that is copied over to /share/ for possible reuse and the install is over this new path.

Wouldn't using installManPage / installShellCompletion there cause duplication?

Comment (SuperSandro2000) [reply pkgs/development/tools/infisical/default.nix:38]: I am not sure where duplications should occur here

File: pkgs/development/tools/infisical/default.nix [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    src = fetchurl {
+      url = "https://github.com/Infisical/infisical/releases/download/v${finalAttrs.version}/infisical_${finalAttrs.version}_${plat}.tar.gz";
+      inherit sha256;
+    };
+
+    sourceRoot = ".";
+    installPhase = ''
+      mkdir -p $out/bin/
+      mkdir -p $out/share/completions/
+      mkdir -p $out/share/man/
```
Comment (SuperSandro2000) [inline pkgs/development/tools/infisical/default.nix:36]:
```suggestion
      mkdir -p $out/bin/ $out/share/completions/ $out/share/man/
```

**Pull request merged**